src/main/java/org/mifos/connector/common/interceptor/JWSUtil.java

```diff
@@ -154,36 +154,67 @@ public static String parseBodyPayload(HttpServletRequest request) throws IOExcep
         return data;
     }
 
+    // parses the boundary value form the content type
+    // works if content type is of below kind
+    // 'multipart/form-data; boundary=--------------------------188270859567880981670528'
+    public static String parseBoundaryValueFromContentTypeValue(String contentType) {
+        String[] contentTypeArray = contentType.split("boundary=");
+        if (contentTypeArray.length < 2) {
+            return "";
+        }
+        return contentTypeArray[1].strip();
+    }
+
     /**
      * Use to parse the string data from the multipart data passed in api
      *
-     * @param httpServletRequest
+     * @param request
      *            request object passed to the controller
      * @return multipart data in form of string
-     * @throws ServletException
      * @throws IOException
      */
-    public static String parseFormData(HttpServletRequest httpServletRequest) throws ServletException, IOException {
+    public static String parseFormData(HttpServletRequest request) throws IOException {
         log.debug("Parsing form data");
-        StringBuilder stringBuilder = new StringBuilder();
-        Collection<Part> parts;
-        try {
-            parts = httpServletRequest.getParts();
-            if (parts == null || parts.isEmpty()) {
-                return "";
+        String contentTypeHeaderValue = request.getHeader(CONTENT_TYPE);
+        String boundary = parseBoundaryValueFromContentTypeValue(contentTypeHeaderValue);
+        return parseFormData(request.getInputStream(), boundary);
+    }
+
+    /**
+     * Use to parse the multipart/form-data  from the input stream
+     *
+     * @param inputStream
+     * @param boundary the boundary can be parsed from CONTENT-TYPE header in
+     *                 API request header
+     * @return parsed data in the form of String
+     * @throws IOException
+     */
+    public static String parseFormData(InputStream inputStream, String boundary) throws IOException {
+        log.debug("inside MultipartParser");
+        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+        StringBuilder partContent = new StringBuilder();
+        String line;
+        while ((line = reader.readLine()) != null) {
+            if (line.contains(boundary)) {
+                // Read and process the part headers
+                while ((line = reader.readLine()) != null && !line.isEmpty()) {
+                    // Process part headers if needed
+                }
+
+                // Read and process the part content
+                while ((line = reader.readLine()) != null && !line.equals(boundary)
+                        && !line.contains(boundary) && !line.equals("\n") && !line.isEmpty()) {
```

Comment:
The loop condition is complex and affects readability of the code. Can you rewrite this?

Reply:
Some of these conditions should be inside the loop. Not as control expression. Code will be much more clear and  readable